### ISSUE TITLE
Provide spatie/period methods as a CarbonPeriod mixin

### ISSUE THREAD

**Issue by kylekatarnls** [briannesbitt/Carbon#1774]
https://github.com/spatie/period contain awesome features to compare and test periods, their period objects are pretty close to `DatePeriod` and so `CarbonPeriod`, so most of their methods could be directly be available on `CarbonPeriod`, for example:

```php
CarbonPeriod::macro('touches', function ($period, ...$arguments) {
  if (!($period instanceof Period)) {
    $period = $period instanceof DatePetiod || $period instanceof CarbonPeriod
      ? Period::make($period->getStartDate(), $period->getEndDate())
      : CarbonPeriod::create($period, ...$arguments);
  }

  return Period::make($this->getStartDate(), $this->getEndDate())->touchesWith($period);
});

CarbonPeriod::create('2010-06-14', '2010-07-18')->touches('2010-06-19', '2010-07-30');
```

Some fundamental methods like equal/contains should be in CarbonPeriod (and should not use the precision mask as in spatie/period) but for other method a good thing would be to have a mixin library users can optionally install, this library would requires spatie/period and provide a mixin class. It could also use #1688 when ready.

It's not required to implement all the method, they could be added incrementally.

If you're interested in creating a library for this, please tell me, I will help if you need and I will mention the library in the Carbon documentation.

**Comment by kylekatarnls** [comment]
After rethinking this issue. If we apply directly certain methods from spatie/period, it can be misleading as CarbonPeriod date interval can be composite (multiple unit, any quantities), and start does not depends on this interval (interval can be 1 day, while start is 2019-08-03 12:30), while spatie/period comparison rely on a precision mask, that is always 1 unit (second/minute/hour/day) so for a precision mask of 1 day, starts is always considered as midnight, so when doing comparisons on period with 1-day-precision, any time of a given day will return the same result for a comparison.

That would not be correct for CarbonPeriod objects.

So we should rather implement our own comparison stuff, eventually methods to get similar roundings than in spatie/period but it would be up to the user to apply it.

Then the mixin library should just provide converters `Carbon\CarbonPeriod > Spatie\Period\Period` and the reverse.

**Comment by kylekatarnls** [comment]
[cmixin/enhanced-period](https://github.com/kylekatarnls/enhanced-period)

It requires at least Carbon 2.23 (released as a beta) and PHP 7.2. It works with both spatie/period 1 and 2.
